hooks/shotgun_launch_shotgun_app.py

```diff
@@ -26,57 +26,54 @@
 
 HookBaseClass = sgtk.get_hook_baseclass()
 
-class LaunchAssociatedApp(HookBaseClass):
-    def execute(self, path, context, associated_entity, **kwargs):
+
+class LaunchShotgunApp(HookBaseClass):
+    def execute(self, published_file, **kwargs):
         """
         Launches the associated app and starts tank.
-        
+
         :param path: full path to the published file
         :param context: context object representing the publish
         :param associated_entity: same as context.entity
         """
-        status = False
-
-        if context is None:
-            raise TankError("Context cannot be None!")
-
         ########################################################################
         # Example implementation below:
+        path = self.get_publish_path(published_file)
+        valid_photoshop_extensions = [".psd", ".jpg", ".jpeg", ".png", ".tiff", ".tga"]
 
+        if published_file.get("task"):
+            context = self.tank.context_from_entity("Task", published_file["task"].get("id"))
+        else:
+            context = self.tank.context_from_path(path)
+        if context is None:
+            raise TankError("Context cannot be None!")
         if path.endswith(".nk"):
             # nuke
-            status = True
             self._do_launch("launchnuke", "tk-nuke", path, context)
-
+            return
         elif path.endswith(".ma") or path.endswith(".mb"):
             # maya
-            status = True
             self._do_launch("launchmaya", "tk-maya", path, context)
-
+            return
         elif path.endswith(".fbx"):
             # Motionbuilder
-            status = True
-            self._do_launch("launchmotionbuilder", "tk-motionbuilder", path, context)            
-            
+            self._do_launch("launchmotionbuilder", "tk-motionbuilder", path, context)
+            return
         elif path.endswith(".hrox"):
             # Hiero
-            status = True
-            self._do_launch("launchhiero", "tk-hiero", path, context)            
-            
+            self._do_launch("launchhiero", "tk-hiero", path, context)
+            return
         elif path.endswith(".max"):
             # 3ds Max
-            status = True
             self._do_launch("launch3dsmax", "tk-3dsmaxplus", path, context)
-            
-        elif path.endswith(".psd"):
+            return
+        elif os.path.splitext(path)[1] in valid_photoshop_extensions:
```

Comment:
Seems to be the only place where `valid_photoshop_extensions` is used, is it worth keeping the variable?